src/export/digitalPlanning/model.ts

```diff
@@ -254,5 +429,273 @@ export class DigitalPlanning {
+            };
+          },
+        ),
+      );
+
+      if (this.passport.data?.["proposal.vehicleParking"][0] !== "none") {
+        const vehicleTypes = [
+          "cars",
+          "vans",
+          "motorcycles",
+          "bicycles",
+          "buses",
+        ];
+        vehicleTypes.forEach((vehicle) => {
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.count.existing`,
+            this.passport.data?.[`proposal.${vehicle}.number.existing`] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.count.proposed`,
+            this.passport.data?.[`proposal.${vehicle}.number.proposed`] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.offStreet.count.existing`,
+            this.passport.data?.[
+              `proposal.${vehicle}.offStreet.number.existing`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.offStreet.count.proposed`,
+            this.passport.data?.[
+              `proposal.${vehicle}.offStreet.number.proposed`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.onStreet.count.existing`,
+            this.passport.data?.[
+              `proposal.${vehicle}.onStreet.number.existing`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.${vehicle}.onStreet.count.proposed`,
+            this.passport.data?.[
+              `proposal.${vehicle}.onStreet.number.proposed`
+            ] || 0,
+          );
+        });
+
+        const carTypes = ["club", "disabled", "other", "residents"];
+        carTypes.forEach((type) => {
+          set(
+            baseProposal,
+            `details.vehicleParking.cars.offStreet.${type}.count.existing`,
+            this.passport.data?.[
+              `proposal.cars.offStreet.${type}.number.existing`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.cars.offStreet.${type}.count.proposed`,
+            this.passport.data?.[
+              `proposal.cars.offStreet.${type}.number.proposed`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.cars.onStreet.${type}.count.existing`,
+            this.passport.data?.[
+              `proposal.cars.onStreet.${type}.number.existing`
+            ] || 0,
+          );
+          set(
+            baseProposal,
+            `details.vehicleParking.cars.onStreet.${type}.count.proposed`,
+            this.passport.data?.[
+              `proposal.cars.onStreet.${type}.number.proposed`
+            ] || 0,
+          );
+        });
+      }
+    }
+
+    return baseProposal;
+  }
+
+  private getFiles(): Payload["files"] {
+    const files: any = [];
```

Comment:
This is similar to the above comment - declaring `const files: File[] = [];` here seems worthwhile. We'll need to cast in the function somewhere but it's more likely we'll catch issues by declaring our intentions here rather than just forcing the return type.

Reply:
I've added a `@todo` code comment to pick up next - I've spent quite a bit of time on this but getting lots of errors still and needing to do a lot of tedious casting, which feels like I'm missing something / a better way.